### PR TITLE
Added IAR problem matcher (#10054)

### PR DESCRIPTION
**Pull request by morsisko**
This pull requests adds IAR problem matcher.

Warning example that is generated by IAR compiler:

`"D:\iar_test\main.c",6  Warning[Pe177]: variable "c" was declared but never referenced`

There is one thing that needs to be said (not sure if it is issue or not). By default for some reasons IAR splits the error/warning message over multiple lines, if the message is too long. In such case problem matcher will catch only the first part of the message, for example:
Code_2022-10-30_23-45-24

However, there is an option that can be passed to IAR - `--no_wrap_diagnostics` in such situation the problem matcher works without any problem.
Code_2022-10-31_00-49-50

I tried to use the multiline problem matcher, however this doesn't look good in my opinion, plus from what I see the IAR message may be split over multiple lines (more than two), so it would be hard or even impossible to implement such problem matcher.

Code_2022-10-30_23-46-12

In my opinion best idea is to leave it as in my pull request. If someone want to get the whole message, they should use the `--no_wrap_diagnostics` IAR option.

### CODE REVIEW

File: Extension/package.json [hunk 1]
```diff
@@ -280,6 +280,19 @@
...
           "message": 5
         }
+      },
+      {
+        "name": "iar",
+        "source": "iar",
+        "owner": "cpptools",
+        "fileLocation" : "absolute",
+        "pattern": {
+          "regexp": "^\"(.*?)\",(\\d+)\\s+(?:[Ff]atal\\s+)?([Ww]arning|[Ee]rror)(\\[\\w+\\d+]:\\s+.*)$",
```
Comment (sean-mcmanus) [inline Extension/package.json]: Seems like `]` should be `\\]`?

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

File: Extension/package.json (hunk 1 above)
Comment (sean-mcmanus) [inline Extension/package.json]: `(\\[\\w+\\d+]:\\s+.*)` can be `\\[(\\w+\\d+)\\]:\\s+(.*)` if you use

```json
"code": 4,
"message": 5
```

to enable displaying the error code separately from the message.

See the C++/cl.exe./mscompile problemMatcher in vscode as an example: https://github.com/microsoft/vscode/blob/c07ff2eb1d8d024accf1f5deba8b6bce384d4cb6/src/vs/workbench/contrib/tasks/common/problemMatcher.ts#L1209

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

File: Extension/package.json (hunk 1 above)
Comment (morsisko) [reply Extension/package.json]: Thanks, you are right

Comment (morsisko) [reply Extension/package.json]: Done

**Comment by morsisko** [comment]
Requested changes has been implemented

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**